strange_function.py

```diff
@@ -0,0 +1,34 @@
+# HAcker Rank Problem: https://www.hackerrank.com/challenges/the-strange-function/problem
+#!/bin/python3
+
+import sys
+
+def find_gcd(x, y): 
+    while(y):
+        x, y = y, x % y
+     
+    return x
+
+def f(a,l,r):
+    gcd_list = a[l]
+    for i in a[l+1: r+1]:
+        gcd_list = find_gcd(gcd_list , i)
+    return gcd_list *(sum(a[l:r+1]) - max(a[l:r+1]))

```

Comment:
The entire for-loop above should be a helper function.

This would simplify this line to:
return gcd(a[l:r+1]) * (sum(a[l:r+1]) - max(a[l:r+1]))

Reply:
Hi Julien,

I have updated the code according to the comments. If you find some time, can you pll
[strange_function.txt](https://github.com/priyaSNHU/Problem-Solving/files/1947204/strange_function.txt)
ease review the code.